public/components/hooks/use_polling.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { useState, useEffect, useRef } from 'react';
+
+interface UsePollingProps<T> {
+  fetchFunction: () => Promise<T>;
```

Comment:
might be better to have fetch function optionally take an abortcontroller, so hook can cancel network requests  when it unmounted